lib/datadog/core/diagnostics/environment_logger.rb

```diff
@@ -5,280 +5,175 @@
 module Datadog
```

Comment:
I recommend [adding `# frozen_string_literal: true`](https://docs.ruby-lang.org/en/3.2/syntax/comments_rdoc.html#label-frozen_string_literal+Directive) to the top of all new Ruby files created in this PR.

This means all strings defined in code are frozen by default, which is very likely we want for new files.

Reply:
makes sense.. done!